cmake/modules/configuration_files.cmake

```diff
@@ -91,9 +100,14 @@ if(NOT DEFINED DTC_OVERLAY_FILE)
                 NAMES "app.overlay" SUFFIX ${FILE_SUFFIX}
     )
   endif()
+endif()
 
+if(DEFINED soc_overlay_files)
+  list(APPEND DTC_OVERLAY_FILE ${soc_overlay_files})
```

Comment:
This actually messes up when qualifier overlays are being applied, because here:
https://github.com/zephyrproject-rtos/zephyr/blob/af7b8400311cb903739554edc059d3592d6e168b/cmake/modules/configuration_files.cmake#L89
they are prepended to board specific overlays, but at this location:
https://github.com/zephyrproject-rtos/zephyr/blob/af7b8400311cb903739554edc059d3592d6e168b/cmake/modules/configuration_files.cmake#L106

they are appended to board overlays discovered here:
https://github.com/zephyrproject-rtos/zephyr/blob/af7b8400311cb903739554edc059d3592d6e168b/cmake/modules/configuration_files.cmake#L96

To keep both implementation and docs as simple as possible, let's follow how overlays are usually picked up, just with the smal adjustment that qualifier are used together with a board overlay from `boards/` folder, but no other overlays.
Similar to how revision overlays are used with board overlays from `board/` folder only.

Reply:
Have moved the soc overlay loading to the board overlay loading and prevented it loading app.overlay